modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/gc/io/GarbageCollectionIo.java

```diff
@@ -0,0 +1,132 @@
+
+    /**
+     * Returns an offset of the element inside the page.
+     *
+     * @see BplusIo#offset(int)
+     */
+    int offset(int idx);
+
+    /**
+     * Stores a row version for garbage collection, copied from another page.
+     *
+     * @see BplusIo#store(long, int, BplusIo, long, int)
+     */
+    default void store(long dstPageAddr, int dstIdx, BplusIo<GarbageCollectionRowVersion> srcIo, long srcPageAddr, int srcIdx) {
+        int dstOffset = offset(dstIdx);
+        int srcOffset = offset(srcIdx);
+
+        PageUtils.copyMemory(srcPageAddr, srcOffset, dstPageAddr, dstOffset, SIZE_IN_BYTES);
+    }
+
+    /**
+     * Stores a row version for garbage collection chain in the page.
+     *
+     * @see BplusIo#storeByOffset(long, int, Object)
+     */
+    default void storeByOffset(long pageAddr, int off, GarbageCollectionRowVersion row) {
+        RowId rowId = row.getRowId();
+
+        putLong(pageAddr, off + ROW_ID_MSB_OFFSET, rowId.mostSignificantBits());
+        putLong(pageAddr, off + ROW_ID_LSB_OFFSET, rowId.leastSignificantBits());
+
+        HybridTimestamps.writeTimestampToMemory(pageAddr, off + ROW_TIMESTAMP_OFFSET, row.getTimestamp());
+    }
+
+    /**
+     * Compare the row version for garbage collection from the page with passed row version, thus defining the order of element in the
+     * {@link GarbageCollectionTree}.
+     *
+     * @param pageAddr Page address.
+     * @param idx Element's index.
+     * @param rowVersion Row version for garbage collection.
+     * @return Comparison result.
+     */
+    default int compare(long pageAddr, int idx, GarbageCollectionRowVersion rowVersion) {
+        int offset = offset(idx);
+
+        int cmp = HybridTimestamps.readTimestamp(pageAddr, offset + ROW_TIMESTAMP_OFFSET).compareTo(rowVersion.getTimestamp());
+
+        if (cmp != 0) {
+            return cmp;
+        }
+
+        RowId rowId = rowVersion.getRowId();
+
+        cmp = Long.compare(getLong(pageAddr, offset + ROW_ID_MSB_OFFSET), rowId.mostSignificantBits());
+
+        if (cmp != 0) {
+            return cmp;
+        }
+
+        return Long.compare(getLong(pageAddr, offset + ROW_ID_LSB_OFFSET), rowId.leastSignificantBits());
+    }
+
+    /**
+     * Reads a row version for garbage collection from the page.
+     *
+     * @param pageAddr Page address.
+     * @param idx Element's index.
+     * @param partitionId Partition id to enrich read partitionless links.
+     */
+    default GarbageCollectionRowVersion getRow(long pageAddr, int idx, int partitionId) {
+        int offset = offset(idx);
+
+        long rowIdMsb = getLong(pageAddr, offset + ROW_ID_MSB_OFFSET);
+        long rowIdLsb = getLong(pageAddr, offset + ROW_ID_LSB_OFFSET);
+
+        return new GarbageCollectionRowVersion(
+                new RowId(partitionId, rowIdMsb, rowIdLsb),
+                HybridTimestamps.readTimestamp(pageAddr, offset + ROW_TIMESTAMP_OFFSET)
```

Comment:
`readTimestamp` can return `null`, is that ok?

Reply:
Fix it